src/test/java/com/alkemy/ong/bigtest/category/CreateCategoryIntegrationTest.java

```diff
@@ -0,0 +1,113 @@
+        .andExpect(status().isCreated()).andReturn();
+    CreateCategoryResponse response = buildResponse(result);
+    assertCategoryHasBeenCreated(response.getId());
+  }
+
+  @Test
+  public void shouldReturnBadRequestWhenNameIsNull() throws Exception {
+    mockMvc.perform(post(URL)
+            .content(buildRequest(buildCreateCategoryRequest(null)))
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .header(HttpHeaders.AUTHORIZATION, getAuthorizationTokenForAdminUser()))
+        .andExpect(jsonPath("$.statusCode", equalTo(400)))
+        .andExpect(jsonPath("$.message", equalTo(INVALID_INPUT_DATA_MESSAGE)))
+        .andExpect(jsonPath("$.moreInfo", hasSize(2)))
+        .andExpect(jsonPath("$.moreInfo", Matchers.hasItem("Name cannot be empty.")))
+        .andExpect(jsonPath("$.moreInfo", Matchers.hasItem("Name must contain only spaces and letters.")))
+        .andExpect(status().isBadRequest());
+  }
+
+  @Test
+  public void shouldReturnBadRequestWhenNameHasNumbers() throws Exception {
+    mockMvc.perform(post(URL)
+            .content(buildRequest(buildCreateCategoryRequest("Category Name 2")))
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .header(HttpHeaders.AUTHORIZATION, getAuthorizationTokenForAdminUser()))
+        .andExpect(jsonPath("$.statusCode", equalTo(400)))
+        .andExpect(jsonPath("$.message", equalTo(INVALID_INPUT_DATA_MESSAGE)))
+        .andExpect(jsonPath("$.moreInfo", hasSize(1)))
+        .andExpect(jsonPath("$.moreInfo", Matchers.hasItem("Name must contain only spaces and letters.")))
+        .andExpect(status().isBadRequest());
+  }
+
+  @Test
+  public void shouldReturnForbiddenWhenUserHasStandardRole() throws Exception {
+    mockMvc.perform(post(URL)
+            .content(buildRequest(buildCreateCategoryRequest(CATEGORY_NAME)))
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .header(HttpHeaders.AUTHORIZATION, getAuthorizationTokenForStandardUser()))
+        .andExpect(jsonPath("$.statusCode", IsEqual.equalTo(403)))
+        .andExpect(jsonPath("$.message", IsEqual.equalTo(ACCESS_DENIED_MESSAGE)))
+        .andExpect(jsonPath("$.moreInfo", hasSize(1)))
+        .andExpect(jsonPath("$.moreInfo", hasItem(ACCESS_DENIED_MORE_INFO)))
+        .andExpect(status().isForbidden());
+  }
+
+  private String buildRequest(CreateCategoryRequest categoryRequest)
+      throws JsonProcessingException {
+    return convert(categoryRequest);
+  }
+
+  private CreateCategoryResponse buildResponse(MvcResult result)
+      throws UnsupportedEncodingException, JsonProcessingException {
+    String jsonString = result.getResponse().getContentAsString();
+    return (CreateCategoryResponse) convert(jsonString, CreateCategoryResponse.class);
+  }
+
+  private void assertCategoryHasBeenCreated(Long createdCategoryId) {
+    Optional<CategoryEntity> category = categoryRepository.exists(createdCategoryId);
```

Comment:
Es preferible utilizar el findById.

Este método fue agregado durante nuestro primer Spring, pero no tiene una firma apropiada, exist/is/has/have/are... something son métodos que deben devolver un valor booleano, no un object..